src/Decompiler/Evaluation/SliceSequence.cs

```diff
@@ -64,9 +63,22 @@ public bool Match(Slice slice)
 
         public Expression Transform()
         {
-            ctx.RemoveIdentifierUse(idOld);
+            ctx.RemoveExpressionUse(eOld);
             ctx.UseExpression(result);
             return result;
         }
+
+        private bool IsSequence(Expression e, out MkSequence sequence)
```

Comment:
I'm thinking that possibly this could be generalized, the way `DataType.ResolveAs`, but I don't think the PR needs to be held up on that account. If this happens often enough, we should consider exposing a `ResolveAs<TExpression>` method on the `EvaluationContext` interface.